fairseq/sequence_generator.py

```diff
@@ -521,9 +526,10 @@ def _decode_one(self, tokens, model, encoder_out, incremental_states, log_probs)
                 decoder_out = list(model.decoder(tokens, encoder_out, incremental_states[model]))
             else:
                 decoder_out = list(model.decoder(tokens, encoder_out))
-            decoder_out[0] = decoder_out[0][:, -1, :]
             attn = decoder_out[1]
             if attn is not None:
                 attn = attn[:, -1, :]
         probs = model.get_normalized_probs(decoder_out, log_probs=log_probs)
+        probs = probs[:, -1, :]
```

Comment:
why move it here? its probably much faster to get probs just for the last state than for everything, and then take the last one?